app/src/main/java/com/example/mytestproject/ui/searchCity/SearchCityViewModel.kt

```diff
@@ -5,35 +5,96 @@ import android.util.Log
 import androidx.lifecycle.LiveData
 import androidx.lifecycle.MutableLiveData
 import androidx.lifecycle.ViewModel
+import com.example.data.db.dao.CityDao
+import com.example.data.mappers.LastTenChosenCitiesEntityMapper
 import com.example.domain.models.CityModel
+import com.example.domain.useCase.weatherData.GetLastTenChosenCitiesUseCase
 import com.example.mytestproject.util.CITY_ID
 import com.example.mytestproject.util.CityFilter
 import com.example.mytestproject.util.Event
+import com.example.mytestproject.util.MAX_TABLE_SIZE
+import io.reactivex.Single
+import io.reactivex.android.schedulers.AndroidSchedulers
 import io.reactivex.disposables.Disposable
+import io.reactivex.schedulers.Schedulers
 
 class SearchCityViewModel(
     private val cityFilter: CityFilter,
-    private val sharePref: SharedPreferences
+    private val sharePref: SharedPreferences,
+    private val cityDao: CityDao,
+    private val lastTenChosenCitiesEntityMapper: LastTenChosenCitiesEntityMapper,
+    private val getLastTenChosenCitiesUseCase: GetLastTenChosenCitiesUseCase
 ) : ViewModel() {
 
+    private var disposable: Disposable? = null
+
+    private val _lastTenChosenCities = MutableLiveData<List<CityModel>>()
+    val lastTenChosenCities: LiveData<List<CityModel>> get() = _lastTenChosenCities
+
     private val _navigateToCurrentWeather = MutableLiveData<Event<Int>>()
     val navigateToCurrentWeather: LiveData<Event<Int>> get() = _navigateToCurrentWeather
 
     private val _filteredCityList = MutableLiveData<List<CityModel>>()
     val filteredCityList: LiveData<List<CityModel>> get() = _filteredCityList
 
+    init {
+        getLastTenChosenCities()
+    }
+
     fun searchCity(inputResult: String) {   //  method sends filtered list of "CityModel"  by user-entered characters to adapter
         _filteredCityList.value = cityFilter.filterCityList(inputResult)
     }
 
     fun onCityChosen(cityId: Int) {
         _navigateToCurrentWeather.value = Event(cityId)
-
         saveCityId(cityId)
+        insertCityToEntity(cityId)
     }
 
     private fun saveCityId(cityId: Int) { // The method  saves the city id to SharedPreferences
         sharePref.edit().putInt(CITY_ID, cityId).apply()
     }
 
+    private fun insertCityToEntity(cityId: Int) { // This method inserts chosen city to db table
```

Comment:
All this logic should be implemented in the Repository.

Also, you are doing a few operations with cityDao. But data in the database can be changed during this time. You need to use transactions here. Room supports it.

Reply:
Fixed